src/main/java/seedu/address/logic/commands/ListCommand.java

```diff
@@ -1,24 +1,73 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
 
+import java.util.stream.Collectors;
+
+import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.model.Model;
+import seedu.address.model.person.ListCommandPredicate;
 
 /**
- * Lists all persons in the address book to the user.
+ * Lists students in the TA Toolkit to the user.
  */
 public class ListCommand extends Command {
 
-    public static final String COMMAND_WORD = "list";
+    public static final String COMMAND_WORD = "ls";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": List all students or students from a specific "
+            + "class group given the class group identifier and displays them as a list with index numbers.\n"
+            + "Parameters: [CLASSGROUP_ID] (must be a valid class group identifier)\n"
+            + "Example: " + COMMAND_WORD + " F14-3";
+
+    public static final String MESSAGE_LIST_ALL_SUCCESS = "Listed all %1$d students";
+
+    public static final String MESSAGE_LIST_CLASS_SUCCESS = "Listed %1$d students from class groups: %2$s";
 
-    public static final String MESSAGE_SUCCESS = "Listed all persons";
+    private final ListCommandPredicate predicate;
+
+    public ListCommand(ListCommandPredicate predicate) {
+        this.predicate = predicate;
+    }
 
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        model.updateFilteredPersonList(predicate);
```

Comment:
Just to check, what is the expected behaviour if ListCommandPredicate is empty?

Reply:
Ah, this is covered in `test()` in ListCommandPredicate.

Reply:
By empty, do you mean the keywords?